frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -339,6 +353,22 @@ describe("QueryBuilder", () => {
   });
 
   describe("beforeunload events", () => {
+    describe("creating models", () => {
```

Comment:
Could you add a test for not showing the confirmation modal when there no changes (isDirty = false)?

Reply:
As explained in PR description:

> New models are always considered dirty, that's why warning will be displayed even if user does not make any changes to an "empty" model

Changing [this behavior](https://github.com/metabase/metabase/blob/ed62130cc557cdabbc163a868b35ba9c88dd3505/frontend/src/metabase-lib/Question.ts#L1099-L1100) would complicate things a bit. We can revisit it later if this warning starts to be annoying (as per [Bruno's suggestion](https://www.notion.so/Prevent-users-from-losing-their-changes-via-Back-and-in-app-navigation-01c80936e86e4196a2750e8a81b8321a?d=9ad5d0f860054cfc83c81863785e2fa7&pvs=4#c24e78de681540c2bac92e519738d8e1)).